### PR TITLE
bpf/tests: test ipv6 udp packets when redirecting from l3 to l2

### PR DESCRIPTION
**Pull request by rgo3**
This PR only changes the respective test to also test udp traffic, to increase the test coverage to ipv4/tcp and ipv6/udp.

### CODE REVIEW

**Review by jschwinger233 — APPROVED** [review]

**Comment by rgo3** [comment]
Hm, locally the test passed. Sometimes I feel like the caching for the BPF integration tests is broken and then a rerun uses previously compiled code :thinking:

File: bpf/tests/tc_nodeport_l3_dev.c [hunk 1]
```diff
@@ -310,24 +310,24 @@ int ipv6_l3_to_l2_fast_redirect_check(__maybe_unused const struct __ctx_buff *ct
...
 
 	if (l4->source != tcp_src_one)
-		test_fatal("src TCP port was changed");
+		test_fatal("l4 src port was changed");
 
 	if (l4->dest != tcp_svc_one)
-		test_fatal("dst TCP port was changed");
+		test_fatal("l4 dst port was changed");
 
 	if (l4->check != bpf_htons(0xdfe3))
```
Comment (jschwinger233) [inline bpf/tests/tc_nodeport_l3_dev.c]: The checksum was hard coded, it should be changed since l4 now is udp.

**Review by jschwinger233 — CHANGES_REQUESTED** [review]

**Review by julianwiedmann — APPROVED** [review]
lgtm, thank you!

**Review by jschwinger233 — APPROVED** [review]

**Comment by rgo3** [comment]
Side quest: why do we have to run `/test` here? Making changes to only `/bpf/tests` feels like an obvious target where it shouldn't be required right?

**Pull request merged**